src/security/Handshake.cc

```diff
@@ -274,39 +274,47 @@ Security::HandshakeParser::parseModernRecord()
     Must(record.fragment.length() || record.type == ContentType::ctApplicationData);
 
     if (currentContentType != record.type) {
+        tkMessages.expectMore(false);
         parseMessages();
         Must(tkMessages.atEnd()); // no currentContentType leftovers
-        fragments = record.fragment;
+        tkMessages.reset(record.fragment, true);
         currentContentType = record.type;
     } else {
-        fragments.append(record.fragment);
+        tkMessages.reset(tkMessages.leftovers().append(record.fragment), true);
```

Comment:
I have a similar concern about unconditional `true` here. AFAICT, both parts of this `if/else` statement share the same "expecting more message bytes" condition (for record.type messages).

Reply:
Done as above.